holidays/countries/united_states.py

```diff
@@ -110,479 +109,869 @@ class UnitedStates(HolidayBase):
-        elif self.subdiv == "GA":
-            if not self._is_wednesday(DEC, 24):
-                self[date(year, DEC, 24)] = name
+        if self._year >= 1879 and self.subdiv not in {
+            "AL",
+            "AR",
+            "DE",
+            "FL",
+            "GA",
+            "NM",
+            "PR",
+            "VI",
+        }:
+            self._add_holiday(
+                "Washington's Birthday",
+                _get_nth_weekday_of_month(3, MON, FEB, self._year)
+                if self._year >= 1971
+                else date(self._year, FEB, 22),
+            )
+
+        # Columbus Day
+        if self._year >= 1937 and self.subdiv not in {
+            "AK",
+            "AR",
+            "DE",
+            "FL",
+            "HI",
+            "NV",
+            "SD",
+            "VI",
+        }:
+            name = "Columbus Day"
+            if self._year >= 1970:
+                self._add_holiday(name, _get_nth_weekday_of_month(2, MON, OCT, self._year))
             else:
-                self[date(year, DEC, 26)] = name
-        elif self.subdiv in {"PR", "VI"}:
-            self[_get_nth_weekday_of_month(3, MON, FEB, year)] = name
+                self._add_columbus_day(name)
 
-        easter_date = easter(year)
-        # Mardi Gras
-        if self.subdiv == "LA" and year >= 1857:
-            self[easter_date + td(days=-47)] = "Mardi Gras"
+        super()._add_subdiv_holidays()
 
-        # Guam Discovery Day
-        if self.subdiv == "GU" and year >= 1970:
-            self[_get_nth_weekday_of_month(1, MON, MAR, year)] = "Guam Discovery Day"
+    def _add_subdiv_ak_holidays(self):
+        # Seward's Day
+        if self._year >= 1918:
+            self._add_holiday(
+                "Seward's Day",
+                _get_nth_weekday_of_month(-1, MON, MAR, self._year)
+                if self._year >= 1955
+                else date(self._year, MAR, 30),
+            )
 
-        # Casimir Pulaski Day
-        if self.subdiv == "IL" and year >= 1978:
-            self[_get_nth_weekday_of_month(1, MON, MAR, year)] = "Casimir Pulaski Day"
+        # Alaska Day
+        if self._year >= 1867:
+            self._add_observed(self._add_holiday("Alaska Day", OCT, 18))
 
-        # Texas Independence Day
-        if self.subdiv == "TX" and year >= 1874:
-            self[date(year, MAR, 2)] = "Texas Independence Day"
+    def _add_subdiv_al_holidays(self):
+        # Martin Luther King Jr. Day
+        if self._year >= 1986:
+            self._add_holiday(
+                "Robert E. Lee/Martin Luther King Birthday",
+                _get_nth_weekday_of_month(3, MON, JAN, self._year),
+            )
 
-        # Town Meeting Day
-        if self.subdiv == "VT" and year >= 1800:
-            self[_get_nth_weekday_of_month(1, TUE, MAR, year)] = "Town Meeting Day"
+        # Washington's Birthday
+        self._add_holiday(
+            "George Washington/Thomas Jefferson Birthday",
```

Comment:
```suggestion
            "George Washington & Thomas Jefferson's Birthday",
```
https://personnel.alabama.gov/Downloads/StateHolidays.pdf